packages/mobile-native-mcp-server/src/workflow/magi/prd/nodes/prdMagiInitialization.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { PRDState } from '../metadata.js';
+import { PRDAbstractToolNode } from './prdAbstractToolNode.js';
+import { ToolExecutor } from '../../../nodes/toolExecutor.js';
+import { Logger } from '../../../../logging/logger.js';
+import { ensureMagiSddDirectory } from '../../../../utils/wellKnownDirectory.js';
+
+export class PRDMagiInitializationNode extends PRDAbstractToolNode {
+  constructor(toolExecutor?: ToolExecutor, logger?: Logger) {
+    super('magiInitialization', toolExecutor, logger);
+  }
+
+  execute = (state: PRDState): Partial<PRDState> => {
+    // Initialize state with user input and default values
+    // Only set defaults for fields that don't already exist (to avoid overwriting existing state)
+    const initializedState: Partial<PRDState> = {
+      userInput: state.userInput || {},
+      // Only initialize these if they're not already set
+      ...(state.functionalRequirements === undefined && { functionalRequirements: [] }),
+      ...(state.gapAnalysisScore === undefined && { gapAnalysisScore: 0 }),
+      ...(state.identifiedGaps === undefined && { identifiedGaps: [] }),
+      ...(state.shouldIterate === undefined && { shouldIterate: false }),
+      ...(state.userIterationOverride === undefined && { userIterationOverride: undefined }),
+      ...(state.prdContent === undefined && { prdContent: '' }),
+      ...(state.prdStatus === undefined && {
+        prdStatus: {
+          author: 'PRD Generator',
+          lastModified: new Date().toISOString(),
+          status: 'draft' as const,
+        },
+      }),
+      ...(state.isPrdApproved === undefined && { isPrdApproved: false }),
+    };
+
+    // Extract required values from userInput
+    const projectPath = state.userInput?.projectPath as string;
+    const userUtterance = state.userInput?.userUtterance as string;
+
+    // Validate that projectPath is provided
+    if (!projectPath || !userUtterance) {
+      // NOTE: this should use a more sophisticated input collection tool - waiting for user input extraction tool to be refactored for re-use here.
+
+      throw new Error(
+        'Both projectPath and userUtterance are required but not provided in userInput. userInput should be a JSON object with the following properties: projectPath and userUtterance.'
+      );
+    }
+
+    try {
+      // Ensure the magi-sdd directory exists in the project path
+      const prdWorkspacePath = ensureMagiSddDirectory(projectPath);
```

Comment:
No action for this PR, but we should sync up on specifics for abstracting the "working directory" functionality in the extraction. We'll probably need some tweaking on one end or the other (or both).